src/openms_gui/source/VISUAL/Plot2DCanvas.cpp

```diff
@@ -3067,12 +3057,14 @@ namespace OpenMS
 
   void Plot2DCanvas::mergeIntoLayer(Size i, vector<PeptideIdentification> & peptides)
   {
-    LayerData& layer = layers_.getLayer(i);
-    OPENMS_PRECONDITION(layer.type == LayerData::DT_IDENT, "Plot2DCanvas::mergeIntoLayer(i, peptides) non-identification layer selected");
+    LayerDataBase& layer = layers_.getLayer(i);
+    OPENMS_PRECONDITION(layer.type == LayerDataBase::DT_IDENT, "Plot2DCanvas::mergeIntoLayer(i, peptides) non-identification layer selected");
```

Comment:
Just an observation: probably important to think about how
layer.type relates to the interfaces
There seems to be some redundancy as DT_IDENT implies IPeptideIds.

Reply:
Yes!
One of the last steps is to remove the `type` (at least from external view), so the user can solely use the much cleaner Interfaces which are about to be created.
But for now, I have to leave the type accessible, because without it everything breaks..